lms/djangoapps/oauth_dispatch/views.py

```diff
@@ -36,7 +36,11 @@ def get_adapter(self, request):
         """
         Returns the appropriate adapter based on the OAuth client linked to the request.
         """
-        if dot_models.Application.objects.filter(client_id=self._get_client_id(request)).exists():
+        dot_application = dot_models.Application.objects.filter(client_id=self._get_client_id(request))
+        if dot_application.exists():
+            if isinstance(self, RevokeTokenView):
+                request.POST = request.POST.copy()  # Make a mutable copy of the POST parameters.
+                request.POST['client_secret'] = dot_application[0].client_secret
```

Comment:
Yeah.  This is a bug in the OAuth library.  For context, see:

- https://github.com/evonove/django-oauth-toolkit/issues/244
- https://github.com/evonove/django-oauth-toolkit/issues/242

According to the first link, this should be fixed in oauthlib version 0.9.0.  For public clients, it no longer requires the `client_secret` field to revoke tokens.

So, we should revert this change and update our current version of `oauthlib` (`0.7.2`) to the `oauthlib` version required by DOT (`1.0.3`) as specified at https://github.com/evonove/django-oauth-toolkit/blob/master/setup.py#L55.  I'm not sure why we don't get a version conflict error when we install DOT  (?).   FYI @jcdyer, @clintonb